src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -881,10 +882,31 @@ static void import_texture_raw(int tile) {
         rdp.loaded_texture[rdp.texture_tile[tile].tmem_index].full_image_line_size_bytes;
     uint32_t line_size_bytes = rdp.loaded_texture[rdp.texture_tile[tile].tmem_index].line_size_bytes;
 
-    for (uint32_t i = 0, j = 0; i < num_loaded_bytes; i += line_size_bytes, j += full_image_line_size_bytes) {
-        memcpy(tex_upload_buffer + i, addr + j, line_size_bytes);
+    // Get the resource's true image size
+    uint32_t resource_image_size_bytes = resource->ImageDataSize;
```

Comment:
I'm not happy using the resource pointer directly here. If we can compute the image size from the width/height info instead from the meta data, that would probably be better.

(width * height * 4) does that in the case of Freezards, but I don't want to assume.

Reply:
GetPointerSize on the Resource is the physical size of data that the resource has allocated, from the perspective of what the game needs from the resource.